client/server/lib/logger/test/index.js

```diff
@@ -1,38 +1,78 @@
 /**
- * Internal dependencies
+ * External dependencies
  */
-import { getLogger } from '../index';
-import bunyan from 'bunyan';
+import { mockProcessStdout } from 'jest-mock-process';
+import mockFs from 'mock-fs';
+import fs from 'fs';
 
-/**
- * Mocks `bunyan` so every instance returned by `createLogger` logs to a buffer instead
- * of process.stdout. We can retrieve that buffer with `bunyan.getOutputBuffer()` and make
- * assertions on it.
- */
-jest.mock( 'bunyan', () => {
-	const realBunyan = jest.requireActual( 'bunyan' );
-	const buffer = new realBunyan.RingBuffer( { limit: 100 } );
-
-	return {
-		createLogger: jest.fn( ( options ) => {
-			return realBunyan.createLogger( {
-				...options,
-				streams: [ { type: 'raw', stream: buffer } ],
-			} );
-		} ),
-		getOutputBuffer: () => buffer,
-	};
+let mockStdout;
+let getLogger;
+
+beforeEach( () => {
+	( { getLogger } = require( '../index' ) );
```

Comment:
I think this could be at the beginning of the file just like any other import:
```js
import { getLogger } from '../index';
```
Jest should be smart enough to move the import to some form of `beforeEach` when transforming the test file and make `jest.resetModules()` work.

Reply:
It does fail when I try that.